aiida_cp2k/workchains/binding_energy.py

```diff
@@ -0,0 +1,218 @@
+        spec.input('protocol_yaml',
+                   valid_type=SinglefileData,
+                   required=False,
+                   help='Specify a custom yaml file with the multistage settings (and ignore protocol_tag)')
+        spec.input('protocol_modify',
+                   valid_type=Dict,
+                   default=Dict(dict={}),
+                   required=False,
+                   help='Specify custom settings that overvrite the yaml settings')
+        spec.input('starting_settings_idx',
+                   valid_type=Int,
+                   default=Int(0),
+                   required=False,
+                   help='If idx>0 is chosen, jumps directly to overwrite settings_0 with settings_{idx}')
+
+        # Workchain outline
+        spec.outline(
+            cls.setup,
+            while_(cls.should_run_geo_opt)(
+                cls.run_stage,
+                cls.inspect_and_update_settings_geo_opt,
+            ),
+            cls.run_bsse,
+            cls.results,
+        )
+
+        # Exit codes
+        spec.exit_code(901, 'ERROR_MISSING_INITIAL_SETTINGS',
+                       'Specified starting_settings_idx that is not existing, or any in between 0 and idx is missing')
+        spec.exit_code(902, 'ERROR_NO_MORE_SETTINGS',
+                       'Settings for Stage0 are not ok but there are no more robust settings to try')
+        spec.exit_code(903, 'ERROR_PARSING_OUTPUT',
+                       'Something important was not printed correctly and the parsing of the first calculation failed')
+
+        # Outputs
+        spec.expose_outputs(Cp2kBaseWorkChain, include=['remote_folder'])
+        spec.output('output_structure',
+                    valid_type=StructureData,
+                    required=False,
+                    help='Processed structure (missing if only ENERGY calculation is performed)')
+        spec.output('last_input_parameters',
+                    valid_type=Dict,
+                    required=False,
+                    help='CP2K input parameters used (and possibly working) used in the last stage')
+        spec.output('output_parameters',
+                    valid_type=Dict,
+                    required=False,
+                    help='Output CP2K parameters of all the stages, merged together')
+
+    def setup_multistage(self):
+        """Setup initial parameters."""
+
+        # Store the workchain inputs in context (to be modified later)
+        self.ctx.base_inp = AttributeDict(self.exposed_inputs(Cp2kBaseWorkChain, 'cp2k_base'))
+
+        # Read yaml file selected as SinglefileData or chosen with the tag, and overwrite with custom modifications
+        if 'protocol_yaml' in self.inputs:
+            self.ctx.protocol = yaml.safe_load(self.inputs.protocol_yaml.open())
+        else:
+            thisdir = os.path.dirname(os.path.abspath(__file__))
+            yamlfullpath = os.path.join(thisdir, 'multistage_protocols', self.inputs.protocol_tag.value + '.yaml')
```

Comment:
@yakutovicha 
do you now a more elegant way to this? Such as open(aiida_cp2k.multistage_protocols.{TAG}.yaml) that works also if this work chain is in aiida-lsmo?